bogo/fetch-and-build

```diff
@@ -18,6 +18,7 @@ COMMIT=47a6f5b4bf4d7acd8d5f7d43cb9c94335cec1c60
 git fetch --depth=1 https://boringssl.googlesource.com/boringssl $COMMIT
 git checkout $COMMIT
 patch -p1 < ../patches/testerrormap.diff
+patch -p1 < ../patches/fix-tls13sessionid.diff
```

Comment:
I think setting COMMIT=c5e4538e3b0bcfdda30d2dabecbee952faf54c1c above is a better approach to getting the patch. Are there downsides to this?

Reply:
I'll give this a shot. It's possible that it requires more shim / config changes.

Reply:
There are 58 failing tests:

 * The current shim doesn't understand the -quic flag, failing one test.
 * The current shim doesn't understand many new flags to tests ALPS.
 * Same for new flags to do GREASE `ClientHello`es.
 * There are several new tests failing with different error classifications than expected.
 
 Do you prefer to update in a different PR before accepting this one?

Reply:
My own personal preference, which isn't official in any capacity in this project, is that we do the work that allows us to independently (from this PR) update to the newer version of Bogo. In general, we should avoid patching Bogo when we can. I filed #465 to track updating to the new version of Bogo.